File: app/src/main/java/com/rst2g1/northkite/ui/smedia/PostAdapter.java
package com.rst2g1.northkite.ui.smedia;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.rst2g1.northkite.databinding.ItemPostBinding;
import com.squareup.picasso.Picasso;

import java.util.List;

public class PostAdapter extends RecyclerView.Adapter<PostViewHolder> {

    private List<Post> posts;
    private OnItemClickListener onItemClickListener;

    public PostAdapter(List<Post> posts, OnItemClickListener onItemClickListener) {
        this.posts = posts;
        this.onItemClickListener = onItemClickListener;
    }

    @NonNull
    @Override
    public PostViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        ItemPostBinding binding = ItemPostBinding.inflate(layoutInflater, parent, false);
        return new PostViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull PostViewHolder holder, int position) {
        Post post = posts.get(position);
        holder.bind(post, onItemClickListener);
    }

    @Override
    public int getItemCount() {
        return posts.size();
    }

    public interface OnItemClickListener {
        void onItemClick(Post post);
    }
}

File: app/src/main/java/com/rst2g1/northkite/ui/smedia/PostAdapter.kt
package com.rst2g1.northkite.ui.smedia

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.rst2g1.northkite.R
import com.rst2g1.northkite.databinding.ItemPostBinding
import com.squareup.picasso.Picasso

class PostAdapter(
    private val posts: List<Post>,
    private val listener: OnItemClickListener
) : RecyclerView.Adapter<PostAdapter.PostViewHolder>() {

    interface OnItemClickListener {
        fun onLikeClick(post: Post)
        fun onCommentClick(post: Post)
        fun onShareClick(post: Post)
        fun onBookmarkClick(post: Post)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PostViewHolder {
        val binding = ItemPostBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return PostViewHolder(binding)
    }

    override fun onBindViewHolder(holder: PostViewHolder, position: Int) {
        val post = posts[position]
        holder.bind(post, listener)
    }

    override fun getItemCount() = posts.size

    inner class PostViewHolder(private val binding: ItemPostBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(post: Post, listener: PostAdapter.OnItemClickListener) {
            binding.textUsername.text = post.username
            binding.textlike.text = String.format("%d likes", post.likes)
            binding.textDescription.text = post.description

            Picasso.get().load(post.profilePicUrl).into(binding.imageProfilePic)
            Picasso.get().load(post.postImageUrl).into(binding.imagePost)

            val currentUserID = "test_user_id" // Replace with method to get current user ID
            if (post.likedBy.contains(currentUserID)) {
                binding.imageLike.setImageResource(R.drawable.heart_red) // Set red heart if liked
            } else {
                binding.imageLike.setImageResource(R.drawable.heart_grey) // Set grey heart if not liked
            }

            binding.imageLike.setOnClickListener { listener.onLikeClick(post) }
            binding.imageComment.setOnClickListener { listener.onCommentClick(post) }
            binding.imageShare.setOnClickListener { listener.onShareClick(post) }
            binding.imageBookmark.setOnClickListener { listener.onBookmarkClick(post) }
        }
    }
}
